### ISSUE TITLE
LogisticRegression: Correctly fit/predict class values that are not (0,1) and squeeze the predictions

### ISSUE THREAD

**Issue by khevn** [h2oai/h2o4gpu#187]
* When the class values are not `(0,1)` the model doesn't produce correct results
* Change the shape of predicted array from `(n,1)` to `(n,)`

```
data = load_breast_cancer()
X = data.data
y = data.target

# convert class values to [0,2]
y = y * 2 

# Splitting data into train and test 
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.25, random_state=42) 

# sklearn
clf_sklearn = linear_model.LogisticRegression()
clf_sklearn.fit(X_train, y_train)
y_pred_sklearn = clf_sklearn.predict(X_test)

# h2o
clf_h2o = h2o4gpu.LogisticRegression()
clf_h2o.fit(X_train, y_train)
y_pred_h2o = clf_h2o.predict(X_test)

print(accuracy_score(y_test, y_pred_sklearn))
print(accuracy_score(y_test, y_pred_h2o.squeeze()))
print(y_test)
print(y_pred_sklearn)
print(y_pred_h2o.squeeze())
```

with target classes being `[0,2]`, accuracy of h2o model drops from `~0.95` to `~0.03`, while sklearn results are not affected. 

Changing the shape of returned array from `(n,1)` to `(n,)`, will allow to directly pass it into helper functions, such as `accuracy_score()`

**Comment by khevn** [comment]
It is also true for other GLM models. The predictions of regression models have `(1, n)` shape, need to reshape them too.

**Comment by mdymczyk** [comment]
I thought you already fixed this, @navdeep-G ?

**Comment by navdeep-G** [comment]
@mdymczyk Don't recall fixing this.

**Comment by mdymczyk** [comment]
@navdeep-G hm thought it was very old issue and got resolved but will run the testcase to make sure.
